### ISSUE TITLE
Error while setting up Azure IoT Operations (az iot ops init)

### ISSUE THREAD

**Issue by iitsDelbruegger** [Azure/azure-cli#28133]
### Describe the bug

When following the quickstart instructions here: https://learn.microsoft.com/en-us/azure/iot-operations/get-started/quickstart-deploy?tabs=linux

on a MicroK8s running on a MacBook Pro with M2 processor, we successfully connected the Kubernetes cluster with Azure Arc. However, we cannot initialize IoT Operations because we get an error message that the azure.identity python module is missing.

### Related command

`az iot ops init --cluster tim-cluster-test -g tim-rg-test --kv-id /subscriptions/*****/resourcegroups/tim-rg-test/providers/Microsoft.KeyVault/vaults/tim-kv-test`

### Errors

The error I am getting:
ModuleNotFoundError: No module named 'azure.identity'

```
Command group 'iot ops' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
The command failed with an unexpected error. Here is the traceback:
No module named 'azure.identity'
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
             ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/lib/python3.11/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/tim/.azure/cliextensions/azure-iot-ops/azext_edge/edge/commands_edge.py", line 185, in init
    return deploy(
           ^^^^^^^
  File "/Users/tim/.azure/cliextensions/azure-iot-ops/azext_edge/edge/providers/orchestration/work.py", line 481, in deploy
    return manager.do_work()
           ^^^^^^^^^^^^^^^^^
  File "/Users/tim/.azure/cliextensions/azure-iot-ops/azext_edge/edge/providers/orchestration/work.py", line 156, in do_work
    from .base import (
  File "/Users/tim/.azure/cliextensions/azure-iot-ops/azext_edge/edge/providers/orchestration/base.py", line 20, in <module>
    from ...util.az_client import get_resource_client
  File "/Users/tim/.azure/cliextensions/azure-iot-ops/azext_edge/edge/util/az_client.py", line 7, in <module>
    from azure.identity import AzureCliCredential
ModuleNotFoundError: No module named 'azure.identity'
To check existing issues, please visit: https://github.com/Azure/azure-cli/issues
```

### Issue script & Debug output

<omitted because it does not include more information>

### Expected behavior

The command should configure my cluster to work with Azure IoT Operations.

### Environment Summary

```
az --version
azure-cli                         2.55.0

core                              2.55.0
telemetry                          1.1.0

Extensions:
azure-iot                         0.23.1
azure-iot-ops                    0.2.0b1
connectedk8s                       1.6.2

Dependencies:
msal                            1.24.0b2
azure-mgmt-resource             23.1.0b2

Python location '/opt/homebrew/Cellar/azure-cli/2.55.0/libexec/bin/python'
Extensions directory '/Users/tim/.azure/cliextensions'

Python (Darwin) 3.11.7 (main, Dec  4 2023, 18:10:11) [Clang 15.0.0 (clang-1500.1.0.2.5)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by iitsDelbruegger** [comment]
> Hi @iitsDelbruegger Find similar issue #22192.
> 
> Issue title	ModuleNotFoundError: No module named 'azure.identity' in latest CLI (v2.36.0)
> Create time	2022-04-26
> Comment number	6
> Please confirm if this resolves your issue.

Well that issue might be related, if the current azure iot operations still assumes that the azure-identity module is provided by the azure cli. However, that issue was closed more than one year ago, so I would think that the problem should be fixed by now.

**Comment by c-ryan-k** [comment]
Hey @iitsDelbruegger - thanks for bringing this to our attention.

Our extension does include `azure-identity` as a requirement, but it seems there's an issue that's surfacing through homebrew CLI installs causing the CLI's module directory (which includes other `azure.*` namespaced packages) to be checked first.

The command then fails to find our included `azure-identity` package that resides in the extension's modules directory `~/.azure/cliextension/azure-iot-ops` which is causing the failure here.

We're working on our fix now and should have an updated release this week.

**Comment by c-ryan-k** [comment]
@iitsDelbruegger - we've released an updated extension `v0.2.0b2` that should address this issue.

Could you please update your extension version with `az extension update --name azure-iot-ops` and see if this fixes your issue?

**Comment by vilit1** [comment]
Closing since there has been no response. If you still have issues, feel free to reopen!
